packages/ic-management/src/types/ic-management.params.ts

```diff
@@ -8,7 +12,64 @@ export interface CanisterSettings {
   computeAllocation?: bigint;
 }
 
-export type UpdateSettingsParams = {
+export function toCanisterSettings({
+  controllers,
+  freezingThreshold,
+  memoryAllocation,
+  computeAllocation,
+}: CanisterSettings = {}): canister_settings {
+  return {
+    controllers: toNullable(controllers?.map((c) => Principal.fromText(c))),
+    freezing_threshold: toNullable(freezingThreshold),
+    memory_allocation: toNullable(memoryAllocation),
+    compute_allocation: toNullable(computeAllocation),
+  };
+}
+
+export interface CreateCanisterParams {
+  settings?: CanisterSettings;
+  senderCanisterVerion?: bigint;
+}
+
+export interface UpdateSettingsParams {
   canisterId: Principal;
+  senderCanisterVerion?: bigint;
   settings: CanisterSettings;
-} & Omit<QueryParams, "certified">;
+}
+
+export enum InstallMode {
+  Install,
+  Reinstall,
+  Upgrade,
+}
+
+export function toInstallMode(
```

Comment:
not biggy but we generally write arrow function


```
export const toInstallMode = () => ...
```

Reply:
Updated now, thanks!